vpr/src/util/vpr_utils.cpp

```diff
@@ -2267,27 +2279,79 @@ int get_sub_tile_physical_pin(int sub_tile_index,
     return result->second.pin;
 }
 
+int get_logical_block_physical_sub_tile_index(t_physical_tile_type_ptr physical_tile,
+                                              t_logical_block_type_ptr logical_block) {
+    int sub_tile_index = OPEN;
+    for (const auto& sub_tile : physical_tile->sub_tiles) {
+        auto eq_sites = sub_tile.equivalent_sites;
+        auto it = std::find(eq_sites.begin(), eq_sites.end(), logical_block);
+        if (it != eq_sites.end()) {
+            sub_tile_index = sub_tile.index;
+        }
+    }
+
+    if (sub_tile_index == OPEN) {
+        VPR_ERROR(VPR_ERROR_OTHER,
+                  "Found no instances of logical block type '%s' within physical tile type '%s'. ",
+                  logical_block->name, physical_tile->name);
+    }
+
+    return sub_tile_index;
+}
+
 int get_physical_pin(t_physical_tile_type_ptr physical_tile,
                      t_logical_block_type_ptr logical_block,
                      int pin) {
+    int sub_tile_index = get_logical_block_physical_sub_tile_index(physical_tile, logical_block);
+
+    if (sub_tile_index == OPEN) {
+        VPR_ERROR(VPR_ERROR_OTHER,
+                  "Couldn't find the corresponding physical tile type pin of the logical block type pin %d.",
+                  pin);
+    }
+
+    int sub_tile_physical_pin = get_sub_tile_physical_pin(sub_tile_index, physical_tile, logical_block, pin);
+    return physical_tile->sub_tiles[sub_tile_index].sub_tile_to_tile_pin_indices[sub_tile_physical_pin];
+}
+
+int get_logical_block_physical_sub_tile_index(t_physical_tile_type_ptr physical_tile,
+                                              t_logical_block_type_ptr logical_block,
+                                              int sub_tile_capacity) {
     int sub_tile_index = OPEN;
     for (const auto& sub_tile : physical_tile->sub_tiles) {
         auto eq_sites = sub_tile.equivalent_sites;
         auto it = std::find(eq_sites.begin(), eq_sites.end(), logical_block);
-        if (it != eq_sites.end()) {
+        if (it != eq_sites.end()
+            && (sub_tile.capacity.is_in_range(sub_tile_capacity))) {
             sub_tile_index = sub_tile.index;
+            break;
         }
     }
 
     if (sub_tile_index == OPEN) {
         VPR_ERROR(VPR_ERROR_OTHER,
-                  "Found no instances of logical block type '%s' within physical tile type '%s'. "
+                  "Found no instances of logical block type '%s' within physical tile type '%s'. ",
+                  logical_block->name, physical_tile->name);
+    }
+
+    return sub_tile_index;
+}
+
+int get_physical_pin(t_physical_tile_type_ptr physical_tile,
```

Comment:
I believe that here, you are trying to get the real physical pin, that accounts also for the sub tile location. IIRC, originally, get_physical_pin was called before/during placement, when a sub tile index was not yet assigned.

With this PR, get_physical_pin needs to be called to get the exact physical pin location of the placed block.

IMO it would be good to have another function instead of the overloaded `get_physical_pin`, which would be called sth like `get_post_place_physical_pin`.

Reply:
I agree. Thanks for the feedback. Will make a change.